Kernel/.gitignore

```diff
@@ -7,3 +7,4 @@ _fs_contents
 sync-local.sh
 *.pcap
 eth_null*
+compile_commmands.json
```

Comment:
What's this? Well, OK if it makes your life easier I suppose, but it seems a little arbitrary

Reply:
> What's this?

It's essentially a project file for Emacs, so I get completion, etc. Like the VS Code stuff I've seen in .gitignores in other places.

> Well, OK if it makes your life easier I suppose, but it seems a little arbitrary

I'm not sure how to force exclude a file from git while also being able to keep it between commits and such, without adding it to the gitignore.